deployer/aws-lambda/kevel/index.js

```diff
@@ -0,0 +1,239 @@
+/* global fetch */
+import { createHmac } from "node:crypto";
+
+import { Client } from "@adzerk/decision-sdk";
+import {
+  KEVEL_SITE_ID,
+  KEVEL_NETWORK_ID,
+  SIGN_SECRET,
+  CARBON_ZONE_KEY,
+  FALLBACK_ENABLED,
+  // eslint-disable-next-line n/no-missing-import
+} from "./env.js";
+import cc2ip from "./cc2ip.js";
+
+const siteId = KEVEL_SITE_ID;
+const networkId = KEVEL_NETWORK_ID;
+const client = new Client({ networkId, siteId });
+
+export async function fetchImage(src) {
+  const imageResponse = await fetch(src);
+  const imageBuffer = await imageResponse.arrayBuffer();
+  const contentType = imageResponse.headers.get("content-type");
+  return { buf: imageBuffer, contentType };
+}
+
+function encodeAndSign(s = "") {
+  const hmac = createHmac("sha256", SIGN_SECRET);
+  hmac.update(s);
+  return `${Buffer.from(s, "utf-8").toString("base64")}.${hmac.digest(
+    "base64"
+  )}`;
+}
+
+function decodeAndVerify(tuple = "") {
+  if (tuple === null) {
+    return null;
+  }
+  const [encoded, digest] = tuple.split(".");
+  const s = Buffer.from(encoded, "base64").toString("utf-8");
+  const hmac = createHmac("sha256", SIGN_SECRET);
+  hmac.update(s);
+  if (hmac.digest("base64") == digest) {
+    return s;
+  }
+  return null;
+}
+
+export async function handler(event) {
+  const request = event.Records[0].cf.request;
+  // https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-cloudfront-headers.html
+  const countryHeader = request.headers["cloudfront-viewer-country"];
+  const countryCode = countryHeader ? countryHeader[0].value : "US";
+  const anonymousIp = cc2ip[countryCode] ?? "127.0.0.1";
+
+  const userAgentHeader = request.headers["user-agent"];
+  const userAgent = userAgentHeader ? userAgentHeader[0].value : null;
+
+  let payload = {};
```

Comment:
This seems to be only used in the first branch, so it could be initialized further below.